src/vs/platform/telemetry/common/errorTelemetry.ts

```diff
@@ -87,7 +88,8 @@ export default abstract class BaseErrorTelemetry {
 		}
 
 		// If it's the no telemetry error it doesn't get logged
-		if (ErrorNoTelemetry.isErrorNoTelemetry(err)) {
+		// TOOD @lramos15 hacking in FileOperation error because it's too messy to adopt ErrorNoTelemetry. A better solution should be found
+		if (ErrorNoTelemetry.isErrorNoTelemetry(err) || err instanceof FileOperationError) {
```

Comment:
I think this is actually not so bad, and quite elegant.

Reply:
The downside here is it makes it unclear when looking at the error events whether or not they will be recorded. I liked the error producer deciding rather than us as the consumer